`shap_values` documentation issues and formula calculation gap.

https://shap.readthedocs.io/en/latest/generated/shap.TreeExplainer.html

```
- **XGBoost and LightGBM** (with default ``model_output="raw"``) output
  a single value (raw margin/log-odds), resulting in SHAP values of shape
  ``(#num_samples, #num_features)`` and ``expected_value`` as a scalar.
```
I find this confusing. Where is that **single value** returned? Compared to sklearn, which has a saparate dimension for the 2 output values, XGBoost doesn't have that. So, where is that **single value** embedded somewhere? Could you please clarify on that?

About the assertion of the values:
(1) 
```
Single output: shap_values[i, :].sum() + expected_value = model_output[i]
```
I try this with XGBoost and this is what I observe:
```
prediction: (92, 2)
shap_values: (1, 20), [[ 0.0194523   0.03423912  0.7979535   0.5000503   0.13220659  0.36267897
   0.20176274  0.          0.528874    0.01454332  0.09491263 -0.00805646
  -0.00625417 -0.01392754  0.02101561  0.28726286  0.          0.02308488
   0.17511454  0.5701697 ]]
expected_value: (), 0.18604861199855804

assert shap_values[0, :].sum() + self._shap.expected_value == prediction[0]
```
There is a mismatach in the prediction shape expectation. It turns out to be 2-D array and therefore hit the `ValueError`:
```
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```
Then, I use `.all()` and the assertion fails:
```
    assert (shap_values[0, :].sum() + self._shap.expected_value == prediction[0]).all(), f"{shap_values[0, :].sum()} + {self._shap.expected_value} = {shap_values[0, :].sum() + self._shap.expected_value} != {prediction[0]}"
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
AssertionError: 3.7350831031799316 + 0.18604861199855804 = 3.9211316108703613 != [0.03359491 0.9664051 ]
```
(2)
```
Multiple outputs: shap_values[i, :, j].sum() + expected_value[j] = model_output[i, j]
```
I try it with RandomForest and this is what I observe:
```
prediction: (92, 2)
shap_values: (1, 20, 2), [[[ 2.90799118e-03 -2.90799118e-03]
  [ 3.53860866e-03 -3.53860866e-03]
  [-5.68251720e-02  5.68251720e-02]
  [-1.76995830e-02  1.76995830e-02]
  [-3.07143696e-02  3.07143696e-02]
  [-3.45794055e-02  3.45794055e-02]
  [-1.34771089e-02  1.34771089e-02]
  [-1.51365344e-02  1.51365344e-02]
  [-5.29854709e-02  5.29854709e-02]
  [-1.27568534e-02  1.27568534e-02]
  [-6.46775658e-03  6.46775658e-03]
  [ 1.48352733e-05 -1.48352733e-05]
  [-1.05228013e-03  1.05228013e-03]
  [-1.35747387e-04  1.35747387e-04]
  [-7.69151868e-04  7.69151868e-04]
  [-3.67810239e-02  3.67810239e-02]
  [-3.43849015e-02  3.43849015e-02]
  [ 1.06271567e-04 -1.06271567e-04]
  [-5.39982993e-02  5.39982993e-02]
  [-6.40255000e-02  6.40255000e-02]]]
expected_value: (2,), [0.45735988 0.54264012]

    assert shap_values[0, :, 0].sum() + self._shap.expected_value[0] == prediction[0, 0], f"{shap_values[0, :, 0].sum()} + {self._shap.expected_value[0]} = {shap_values[0, :, 0].sum() + self._shap.expected_value[0]} != {prediction[0, 0]}"
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: -0.42522145177905124 + 0.45735988368679786 = 0.03213843190774662 != 0.13248437592609275
```

--- a/shap/explainers/_tree.py
+++ b/shap/explainers/_tree.py
@@ -550,12 +550,12 @@ class TreeExplainer(Explainer):
             * Single output: ``shap_values[i, :].sum() + expected_value = explainer_model_output[i]``
             * Multiple outputs: ``shap_values[i, :, j].sum() + expected_value[j] = explainer_model_output[i, j]``
 
-            .. note:: 
-               The ``explainer_model_output`` is NOT necessarily what ``model.predict()`` 
-               or ``model.predict_proba()`` returns. For example, for an XGBoost Classifier with the default 
-               ``model_output="raw"``, the explainer returns log-odds (margins). 
+            .. note::
+               The ``explainer_model_output`` is NOT necessarily what ``model.predict()``
+               or ``model.predict_proba()`` returns. For example, for an XGBoost Classifier with the default
+               ``model_output="raw"``, the explainer returns log-odds (margins).
                To compare this mathematically against ``predict_proba()`` probabilities,
-               a logistic inverse-transform (e.g., ``scipy.special.expit``) must be applied 
+               a logistic inverse-transform (e.g., ``scipy.special.expit``) must be applied
                to the sum.
 
             The shape of the returned array depends on the number of model outputs:

--- a/shap/explainers/_tree.py
+++ b/shap/explainers/_tree.py
@@ -904,6 +904,16 @@ class TreeExplainer(Explainer):
                     f" was {sum_val[ind]:f}, while the model output was {model_output[ind]:f}. If this"
                     " difference is acceptable you can set check_additivity=False to disable this check."
                 )
+                # Diagnostic for common binary classification link function confusion (GH #4414)
+                try:
+                    if np.allclose(scipy.special.expit(sum_val), model_output, atol=1e-2, rtol=1e-2):
+                        err_msg += (
+                            "\n\nHINT: The SHAP values sum to the model's raw output (margins), "
+                            "but it looks like you are comparing them to probabilities. "
+                            "Apply a logistic transform (scipy.special.expit) to the sum to match."
+                        )
+                except Exception:
+                    pass
                 raise ExplainerError(err_msg)
 
         if isinstance(phi, list):

--- a/shap/explainers/_tree.py
+++ b/shap/explainers/_tree.py
@@ -543,15 +543,14 @@ class TreeExplainer(Explainer):
         np.array
             Estimated SHAP values, usually of shape ``(# samples x # features)``.
 
-            For each output, the SHAP values (summed across all features) plus the
-            expected value equals the model's output in the space of the ``model_output``
-            argument for that sample:
+            For each output, the sum of the SHAP values plus the ``expected_value``
+            equals the model's output (in the specified output space):
 
-            * Single output: ``shap_values[i, :].sum() + expected_value = explainer_model_output[i]``
-            * Multiple outputs: ``shap_values[i, :, j].sum() + expected_value[j] = explainer_model_output[i, j]``
+            * Single output: ``shap_values[i, :].sum() + expected_value = f(x)[i]``
+            * Multiple outputs: ``shap_values[i, :, j].sum() + expected_value[j] = f(x)[i, j]``
 
             .. note::
-               The ``explainer_model_output`` is NOT necessarily what ``model.predict()``
+               The ``f(x)`` value is NOT necessarily what ``model.predict()``
                or ``model.predict_proba()`` returns. For example, for an XGBoost Classifier with the default
                ``model_output="raw"``, the explainer returns log-odds (margins).
                To compare this mathematically against ``predict_proba()`` probabilities,

--- a/shap/explainers/_tree.py
+++ b/shap/explainers/_tree.py
@@ -544,10 +544,19 @@ class TreeExplainer(Explainer):
             Estimated SHAP values, usually of shape ``(# samples x # features)``.
 
             For each output, the SHAP values (summed across all features) plus the
-            expected value equals the model's output for that sample:
+            expected value equals the model's output in the space of the ``model_output``
+            argument for that sample:
 
-            * Single output: ``shap_values[i, :].sum() + expected_value = model_output[i]``
-            * Multiple outputs: ``shap_values[i, :, j].sum() + expected_value[j] = model_output[i, j]``
+            * Single output: ``shap_values[i, :].sum() + expected_value = explainer_model_output[i]``
+            * Multiple outputs: ``shap_values[i, :, j].sum() + expected_value[j] = explainer_model_output[i, j]``
+
+            .. note:: 
+               The ``explainer_model_output`` is NOT necessarily what ``model.predict()`` 
+               or ``model.predict_proba()`` returns. For example, for an XGBoost Classifier with the default 
+               ``model_output="raw"``, the explainer returns log-odds (margins). 
+               To compare this mathematically against ``predict_proba()`` probabilities,
+               a logistic inverse-transform (e.g., ``scipy.special.expit``) must be applied 
+               to the sum.
 
             The shape of the returned array depends on the number of model outputs:
 

--- a/shap/explainers/_tree.py
+++ b/shap/explainers/_tree.py
@@ -558,6 +558,9 @@ class TreeExplainer(Explainer):
                a logistic inverse-transform (e.g., ``scipy.special.expit``) must be applied
                to the sum.
 
+               Furthermore, the additivity formula requires SHAP values and model
+               predictions to be computed on the same samples in the same order.
+
             The shape of the returned array depends on the number of model outputs:
 
             * one output: array of shape ``(#num_samples, *X.shape[1:])``.

--- a/shap/explainers/_tree.py
+++ b/shap/explainers/_tree.py
@@ -904,14 +904,7 @@ class TreeExplainer(Explainer):
                     f" was {sum_val[ind]:f}, while the model output was {model_output[ind]:f}. If this"
                     " difference is acceptable you can set check_additivity=False to disable this check."
                 )
-                # Diagnostic for common binary classification link function confusion (GH #4414)
-                if sum_val.shape == model_output.shape:
-                    if np.allclose(scipy.special.expit(sum_val), model_output, atol=1e-2, rtol=1e-2):
-                        err_msg += (
-                            "\n\nHINT: The SHAP values sum to the model's raw output (margins), "
-                            "but it looks like you are comparing them to probabilities. "
-                            "Apply a logistic transform (scipy.special.expit) to the sum to match."
-                        )
+
                 raise ExplainerError(err_msg)
 
         if isinstance(phi, list):

--- a/shap/explainers/_tree.py
+++ b/shap/explainers/_tree.py
@@ -905,15 +905,13 @@ class TreeExplainer(Explainer):
                     " difference is acceptable you can set check_additivity=False to disable this check."
                 )
                 # Diagnostic for common binary classification link function confusion (GH #4414)
-                try:
+                if sum_val.shape == model_output.shape:
                     if np.allclose(scipy.special.expit(sum_val), model_output, atol=1e-2, rtol=1e-2):
                         err_msg += (
                             "\n\nHINT: The SHAP values sum to the model's raw output (margins), "
                             "but it looks like you are comparing them to probabilities. "
                             "Apply a logistic transform (scipy.special.expit) to the sum to match."
                         )
-                except Exception:
-                    pass
                 raise ExplainerError(err_msg)
 
         if isinstance(phi, list):

